edx_toggles/toggles/internal/waffle/base.py

```diff
@@ -0,0 +1,35 @@
+"""
+Base waffle toggle classes.
+"""
+
+from edx_django_utils.monitoring import set_custom_attribute
+
+from ..base import BaseToggle
+
+
+# pylint: disable=abstract-method
+class BaseWaffle(BaseToggle):
+    """
+    Base waffle toggle class, which performs waffle name validation.
+    """
+
+    def __init__(self, name, module_name=None):
```

Comment:
Well, it's possible, but I'm not a big fan. As I mentioned elsewhere, the `module_name` is only used to track code owners in the toggle state endpoint. The `module_name` is useless in the primary use case for feature toggles -- which is to actually enable/disable features. So I would be reluctant to make that argument mandatory.

Reply:
I feel reasonably confident that I want this for the following reasons:
* We need it to maintain our system, and it is the least expensive way to ensure it is there. Linting is a possibility, but it is a lot more complicated to make sure linting is happening everywhere you want it to.
* The toggle state endpoint reports the `module_name`, whether or not it is mapped to a code owner.  Could still be useful to anyone for tracking down toggles that come in from plugins, etc.
* Since `__name__` is always available, it basically puts no extra burden on developers.

If you still disagree, I'm happy to have us solicit additional opinions.